dotnet/src/VectorDataIntegrationTests/VectorDataIntegrationTests/CRUD/ConformanceTestsBase.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.Extensions.VectorData;
+using VectorDataSpecificationTests.Support;
+
+namespace VectorDataSpecificationTests.CRUD;
+
+// TKey is a generic parameter because different connectors support different key types.
+public abstract class ConformanceTestsBase<TKey, TRecord>(VectorStoreFixture fixture) where TKey : notnull
+{
+    protected VectorStoreFixture Fixture { get; } = fixture;
+
+    protected virtual string GetUniqueCollectionName() => Guid.NewGuid().ToString();
+
+    protected virtual VectorStoreRecordDefinition? GetRecordDefinition() => null;
+
+    protected async Task ExecuteAsync(Func<IVectorStoreRecordCollection<TKey, TRecord>, Task> test)
+    {
+        string collectionName = this.GetUniqueCollectionName();
+        var collection = this.Fixture.TestStore.DefaultVectorStore.GetCollection<TKey, TRecord>(collectionName,
+            this.GetRecordDefinition());
+
+        await collection.CreateCollectionAsync();
```

Comment:
Unfortunately, creating a collection is a bit more complicated - in some databases this happens asynchronously, so CreateCollectionAsync returns before the collection is actually ready, and the test will fail.

See how this is done in BasicfilterTests - I've tried to design this in a way which is reusable across test suites (e.g. WaitForDataAsync).

Reply:
Thanks, I was not aware of that. Is it OK to fix it later, when we actually add some tests for such DBs?